packages/lesmis/src/api/queries/useDashboardReportData.js

```diff
@@ -5,31 +5,69 @@
  */
 import { useQuery } from 'react-query';
 import { get } from '../api';
+import { useDashboardData } from './useDashboardData';
+import { combineQueries } from './utils';
+import { QUERY_OPTIONS } from './constants';
 
-export const useDashboardReportData = ({
-  entityCode,
-  reportCode,
-  itemCode,
-  dashboardCode,
-  legacy,
-  startDate,
-  endDate,
-}) => {
+const useDashboardReportData = ({ entityCode, reportCode, startDate, endDate }) => {
   const params = {
     startDate,
     endDate,
-    itemCode,
-    dashboardCode,
-    legacy,
     type: 'dashboard',
   };
 
+  const enabled = startDate !== undefined && endDate !== undefined && !!reportCode;
+
   return useQuery(
     ['dashboardReport', entityCode, reportCode, params],
     () =>
       get(`report/${entityCode}/${reportCode}`, {
         params,
       }),
-    { staleTime: 60 * 60 * 1000, refetchOnWindowFocus: false, keepPreviousData: true },
+    { ...QUERY_OPTIONS, keepPreviousData: true, enabled },
+  );
+};
+
+/**
+ * Gets a list of reportCodes keyed by dashboardItemCodes
+ * @param dashboards
+ * @returns {*}
+ */
+const getReportCodesByCode = dashboards => {
+  return dashboards?.reduce((items, dash) => {
+    const newItems = {};
+    dash.items.forEach(item => {
+      newItems[item.code] = item.reportCode;
+    });
+    return { ...items, ...newItems };
+  }, {});
+};
+
+export const useDashboardReportDataWithConfig = ({
+  entityCode,
+  reportCode,
+  startDate,
+  endDate,
+}) => {
+  const query = combineQueries({
+    reportData: useDashboardReportData({ entityCode, reportCode, startDate, endDate }),
+    dashboards: useDashboardData({ entityCode }),
+  });
+
+  const dashboard = query.data?.dashboards?.find(dash =>
+    dash.items.find(item => item.reportCode === reportCode),
   );
+
+  const dashboardItem = dashboard?.items.find(item => item.reportCode === reportCode);
+
+  const reportCodes = getReportCodesByCode(query.data?.dashboards);
```

Comment:
The report code is almost always the same as the dashboardItem code but my understanding is that it is not safe to assume they are the same, right?

Reply:
That's right